### ISSUE TITLE
[nextgen-editor] Unclear How to Manage If/When NextGen is Applied in Blueprints

### ISSUE THREAD

**Issue by thekenshow** [getgrav/grav-premium-issues#188]
I'm unclear how to selectively apply the NextGen editor to page blueprints, and how that interacts with the plugin settings. It seems that:

* When **Default for All == Enabled**,  every markdown and textarea (?)  has NextGen applied
* When **Default for All = Disabled**,  even the default content field does not have NextGen applied

With **Default for All = Disabled**, I tried the custom page blueprint below. I want the default content field to use NextGen and the custom `.desc` field to be a simple textarea. Instead, both show up with markdown editors (same when using `type: editor` for the content field):

```
  fields:
    tabs:
      type: tabs
      active: 1

      fields:
        content:
          type: tab
          title: PLUGIN_ADMIN.CONTENT
          
          fields:
            content:
              type: markdown

            header.blocks:
              type: list
              label: 'Content blocks'
              fields:
                .title:
                  type: text
                  label: Name
                  help: 'Provide a name for this content block'
                  vaidate:
                      required: yes
                .desc:
                  type: textarea
                  label: Text
                  help: 'Provide a plain text description, up to 130 characters'
                  validate:
                      required: yes
```

So (assuming I haven't messed up the blueprint above, which is possible!), I have  two points:

1. I can't see any scenario where a user would purchase and install NextGen and *not want the default content field to use it*. So it should be exempt from the **Default for All = Disabled** scope, and require explicit override to change. 
2. I didn't find any documentation of how nextgen fits into the possible field types:  editor, markdown, textarea. Is there a nextgen? If not, it would be good to explain specifically how it interacts with the existing types and the Default for All setting.

**Comment by thekenshow** [comment]
Related: I have custom flex directories that use textarea fields, and they aren't picking up the nextgen editor when editing in Admin. They work as before when set as "textarea" and "markdown".

**Comment by w00fz** [comment]
Thank you for the detailed issue report @thekenshow - you were right, some oddity was going on with that field.

I have now taken care of this so that the "Default for all **users**" works as expected and only targets the 'content' field, if set as **markdown** type. 

I also added a new custom field `nextgen-editor` which you can freely use independently of the default editor setting.

In conclusion, with your example above, you can now replace `type: markdown` with `type: nextgen-editor` and it wouldn't matter what the settings are for the user editor, it will always use nextgen-editor.

This will be available in the next release, v1.1.8.

Thanks!

**Comment by thekenshow** [comment]
Great, looking forward to v 1.1.8.
